ui/webapp.py

```diff
@@ -1,28 +1,105 @@
 import streamlit as st
 from utils import retrieve_doc
+from utils import feedback_doc
 from annotated_text import annotated_text
+import st_state_patch
+import pandas as pd
+import json
 
 def annotate_answer(answer,context):
     start_idx = context.find(answer)
     end_idx = start_idx+len(answer)
     annotated_text(context[:start_idx],(answer,"ANSWER","#8ef"),context[end_idx:])
-      
+
+def random_questions(df):
+     random_row = df.sample(1)
+     random_question = random_row["Question Text"].values[0]
+     random_answer = random_row["Answer"].values[0]
+     return random_question, random_answer
+
+# Define state
+state_question = st.State()
+state_run_query = st.State()
+
+# Initalize variables
+eval_mode = False
+random_question = "Who is the father of Arya Starck?"
+
+# UI search bar and sidebar      
 st.write("# Haystack Demo")
 st.sidebar.header("Options")
 top_k_reader = st.sidebar.slider("Max. number of answers",min_value=1,max_value=10,value=3,step=1)
 top_k_retriever = st.sidebar.slider("Max. number of documents from retriever",min_value=1,max_value=10,value=3,step=1)
-question = st.text_input("Please provide your query:",value="Who is the father of Arya Starck?")
-run_query = st.button("Run")
+eval_mode = st.sidebar.checkbox("Evalution mode")
 debug = st.sidebar.checkbox("Show debug info")
+
+# load csv into pandas dataframe
+if eval_mode:
+    df = pd.read_csv("eval_labels_example.csv", sep=";")
```

Comment:
This we would need to configure from outside

Reply:
Yes,  there is a file with two example questions and answers. The user has to replace it with his data.

Reply:
K this hard coding of files would be an option. Cant we make it a parameter of our docker-compose file? I would prefer the second option if possible.

And what about the underlying documents? Currently we need to have them pre-indexed in ES. Can we either have a description on how to add the documents to an existing ES or add them in a spearate file like you did with questions?